safehttp/mux.go

```diff
@@ -88,18 +91,25 @@ func NewServeMux(d Dispatcher, domains ...string) *ServeMux {
 		domains:  dm,
 		disp:     d,
 		handlers: map[string]methodHandler{},
+		plugins:  map[string]Plugin{},
 	}
 }
 
 // Handle registers a handler for the given pattern and method. If another
 // handler is already registered for the same pattern and method, Handle panics.
 func (m *ServeMux) Handle(pattern string, method string, h Handler) {
+	hp := handlerWithPlugins{
+		handler:        h,
+		disp:           m.disp,
+		muxPlugins:     m.plugins,
+		handlerPlugins: map[string]Plugin{},
```

Comment:
Try to make your PRs smaller.

In this case, we are likely not going to have per-handler plugins, but per-handler plugin configurations. So drop it for now and just focus on shipping the per-mux ones.

Later, you'll be thinking about:
- per plugin configuration
- ordering
- speedups (avoiding allocations and branches on the hot path)
- plugin configs conformance checks
- and so on.

So, again, keep them small and avoid going two steps ahead.